src/Engine/Party.cpp

```diff
@@ -1175,3 +1175,12 @@ void Party::giveFallDamage(int distance) {
         }
     }
 }
+
+Character *Party::getCharacterByIndex(int characterIndex) {
```

Comment:
I honestly don't like seeing this function in `Party` becase it's essentially lua middleware that's leaking into the codebase. Let's put this into `GameLuaBindings.cpp`.